examples/provisioned.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+
+use Illuminate\Database\DatabaseManager;
+use SaatchiArt\BentoBoxDDD\Adapters\Repositories\ArtworkRepositoryMysqlImpl;
+use SaatchiArt\BentoBoxDDD\Adapters\Repositories\Transactions\TransactionFactory;
+use SaatchiArt\BentoBoxDDD\Adapters\Repositories\UserRepositoryMysqlImpl;
+use SaatchiArt\BentoBoxDDD\Domain\Repositories\ArtworkRepository;
+use SaatchiArt\BentoBoxDDD\Domain\Repositories\UserRepository;
+
+class SomeService {
+    private ArtworkRepository $artworkRepository;
+    private UserRepository $userRepository;
+    private TransactionFactory $transactionFactory;
+
+    public function __construct(
+        ArtworkRepository $artworkRepository,
+        UserRepository $userRepository,
+        TransactionFactory $transactionFactory
+    ) {
+        $this->artworkRepository = $artworkRepository;
+        $this->userRepository = $userRepository;
+        $this->transactionFactory = $transactionFactory;
+    }
+
+    public function updateThoseArtworksForSomeReason(): int
+    {
+        $transaction = $this->transactionFactory->makeFullyProvisionedTransaction(
+            [$this->artworkRepository, $this->userRepository],
+            static function (
+                ArtworkRepository $artworkRepository,
+                UserRepository $userRepository
```

Comment:
Strongly typed anonymous function which MUST match the provisioning above will declare the scope of transaction, and which repos support it. Unless one cheats with a `use ($someOtherRepository)` you are protected.